src/pykeen/triples/statement_factory.py

```diff
@@ -0,0 +1,600 @@
+# -*- coding: utf-8 -*-
+
+"""Implementation of basic instance factory which creates instances of hyper-relational statements."""
+import functools
+import logging
+import pathlib
+import re
+from abc import abstractmethod
+from collections import defaultdict
+from typing import (
+    Any,
+    Callable,
+    ClassVar,
+    Collection,
+    Dict,
+    Iterable,
+    List,
+    Mapping,
+    MutableMapping,
+    Optional,
+    Sequence,
+    Set,
+    TextIO,
+    Tuple,
+    TypeVar,
+    Union,
+    cast,
+)
+
+import numpy as np
+import torch
+from tqdm import tqdm
+
+from .instances import LCWAInstances, SLCWAInstances
+from .triples_factory import KGInfo
+from .utils import STATEMENT_PADDING, load_rdf_star_statements
+from ..typing import EntityMapping, LabeledTriples, MappedTriples, RelationMapping, TorchRandomHint
+from ..utils import (
+    ExtraReprMixin,
+    compact_mapping,
+    format_relative_comparison,
+    get_edge_index,
+    invert_mapping,
+    normalize_path,
+    triple_tensor_to_set,
+)
+
+logger = logging.getLogger(__name__)
+
+INVERSE_SUFFIX = "_inverse"
+TRIPLES_DF_COLUMNS = ("head_id", "head_label", "relation_id", "relation_label", "tail_id", "tail_label")
```

Comment:
These may already exist as constants either in `pykeen/constants.py` or `pykeen/typing.py`